### PR TITLE
Add Travis support and an autotools example

### PR DESCRIPTION
**Pull request by lallemfa**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+TO DO
+
+## Src
+
+This folder contains an example for autotools use in the folder domoticz.
+In order to compile it, run:
+- ./autogen.sh
+- ./configure
+- make
```
Comment (pblottiere) [inline README.md]: In markdown, you should insert bash commands this way:

``` shell
$ ./autogen.sh
$ ./configure
$ make
```

File: configure.ac [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Prelude
+AC_PREREQ(2.4)
+AC_INIT([MyPackageName], [0.1], [blottiere.paul@gmail.com])
```
Comment (pblottiere) [inline configure.ac]: Can you rename the package in `dominus`?

File: configure.ac [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+AC_PROG_CC
+AM_CONFIG_HEADER([config.h])
+
+# libtool init
+LT_INIT
+AC_ENABLE_SHARED
+
+# check headers
+AC_CHECK_HEADER(sys/socket.h, [],
+                [AC_MSG_ERROR(["Couldn't find sys/socket.h"])])
```
Comment (pblottiere) [inline configure.ac]: I don't think you need `socket.h` header.

File: src/domoticz/Makefile.am [hunk 4]
```diff
@@ -0,0 +1,6 @@
+MAINTAINERCLEANFILES = Makefile.in
+
+AM_LDFLAGS = -lcurl
+
+bin_PROGRAMS = createSensor
+createSensor_SOURCES = createSensor.c
```
Comment (pblottiere) [inline src/domoticz/Makefile.am]: Naming convention for C files would be rather something like `createsensor.c` or even `create_sensor.c`.

Comment (pblottiere) [reply src/domoticz/Makefile.am]: BTW, it's true for the binary too

File: src/domoticz/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# DOMINUS - Src domoticz
+
+
+
+## createSensor.c
+
+This code is designed to allow users to easily add sensors to a remote domoticz server.
+Use it as follows:
+
+    ./createSensor IP Port Sensor_type Sensor_name ID_Sensor
```
Comment (pblottiere) [inline src/domoticz/README.md]: Markdown:

``` shell 
$ ./createSensor IP Port Sensor_type Sensor_name ID_Sensor
```

File: src/domoticz/createSensor.c [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        // Will contain the parameters
+        char* IP_address; 
+        char* port;
+        char* name;
+        int ID_Sensor, Type_Sensor;
+
+        char http_address[250]; // buffer for the http request
+
+
+                // CURL Handlers
```
Comment (pblottiere) [inline src/domoticz/createSensor.c]: indentation

File: src/domoticz/createSensor.c [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        CURL *curl;
+        CURLcode res;
+
+
+                // Check correct use and update parameters
```
Comment (pblottiere) [inline src/domoticz/createSensor.c]: indentation

File: src/domoticz/createSensor.c (hunk 6 above)
Comment (pblottiere) [reply src/domoticz/createSensor.c]: And I would prefer 4 spaces for an indentation level :)

File: src/domoticz/createSensor.c [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                                \t249 -> TypeAirQuality\n\n \
+                        Needed 5 arguments but %d given.\nUse this way -> ./createSensor IP Port Sensor_type Sensor_name ID_Sensor\n", argc-1);
+                return EXIT_FAILURE;
+        } else {
+                IP_address      = argv[1];                
+                port            = argv[2];
+                Type_Sensor     = atoi(argv[3]);
+                name            = argv[4];
+                ID_Sensor       = atoi(argv[5]);                
+        }
```
Comment (pblottiere) [inline src/domoticz/createSensor.c]: What's happening if you don't have enough arguments on the command line? Do not forget to always handle errors.

File: src/domoticz/createSensor.c [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+                        
+
+                        // Request 2 -> List of all devices
+
+                sprintf(http_address, "http://%s:%s/json.htm?type=devices&filter=all&used=true&order=Name", IP_address, port);
+                curl_easy_setopt(curl, CURLOPT_URL, http_address);
+                res = curl_easy_perform(curl);
+                if(res != CURLE_OK)
+                {
+                        fprintf(stderr, "curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
```
Comment (pblottiere) [inline src/domoticz/createSensor.c]: An error code should be returned in this case (and not `EXIT_SUCCESS`).

**Comment by pblottiere** [comment]
Hi guys! Thank you for your work :+1:!

I just made some comments on your code. If you don't have time to fix this, let me know and I'll handle it myself.

BTW, you should create and work in a specific branch when you're doing a pull request (I mean, not in `master` ;)).

**Comment by lallemfa** [comment]
Hi !

I would rather prefere to fix this myself in the upcoming days. It would allow me to better see what was wrong in my code.
However, I do not promise to be able to do so in a very short time. Is there a deadline sooner than the next course ? Even if I really doubt not being able to fix this far before this deadline.
Indeed, I will do so for the branch.

Thanks a lot for your comments !

Fabrice



________________________________
De : Blottiere Paul <notifications@github.com>
Envoyé : Tuesday, December 5, 2017 9:42:49 PM
À : pblottiere/dominus
Cc : Fabrice LALLEMENT (CI_2018); Author
Objet : Re: [pblottiere/dominus] Add Travis support and an autotools example (#1)


Hi guys! Thank you for your work 👍!

I just made some comments on your code. If you don't have time to fix this, let me know and I'll handle it myself.

BTW, you should create and work in a specific branch when you're doing a pull request (I mean, not in master ;)).

**Comment by pblottiere** [comment]
> I would rather prefere to fix this myself 

Cool :)

> Is there a deadline sooner than the next course ?

Nope, no pressure ;)

File: src/domoticz/createSensor.c (hunk 8 above)
Comment (lallemfa) [reply src/domoticz/createSensor.c]: I am not sure if I understood what you meant. I think I am checking that there is enough arguments with the first If... Else...
However it's true I am not checking that the parameters are consistents (e.g. the integer for the sensor's type means something for domoticz).

Comment (pblottiere) [reply src/domoticz/createSensor.c]: > I am not sure if I understood what you meant. I think I am checking that there is enough arguments with the first If... Else...

Arf yes, you're right, my bad.

**Comment by pblottiere** [comment]
Nice, thank you for taking into account my comments!

**Pull request merged**